pagopa/pn-bff
# fix/PN-13269: Notification Receipt not translated

## Short description
This PR changes changes the content of otherDocuments available inside the notification detail, setting the title to an empty string and adding a recipient object, containing the recipient denomination and taxid. These changes allow the FE to easily display a translated content for the Notification Receipt and grant a backward compatibility since all the fields previously contained inside the response are still there.

## List of changes proposed in this pull request
- NotificationDetailUtility.java
- NotificationDetailUtilityTest.java

## How to test
Use postman to verify the otherDocuments field contained inside the notification detail has the new recipient object showing recipient's denomination and taxId
